src/buffer/cell.rs

```diff
@@ -67,6 +67,14 @@ impl Cell {
         self
     }
 
+    /// Appends a symbol to the cell.
+    ///
+    /// This is particularly useful for adding zero-width characters to the cell.
+    pub fn append_symbol(&mut self, symbol: &str) -> &mut Self {
```

Comment:
I'm not sure whether I like this complexity being added to this low level building block…

Reply:
I can see that point. What about making this `pub(crate)`?